B/G deployments (#57)
porter-dev/porter-docs: docs/deploying-applications/blue-green-deployments.mdx

```yaml
bluegreen:
  activeImageTag: <tag>
  disablePrimaryDeployment: true
  enabled: true
  imageTags:
    - <tag>
```

For `<tag>`, you'll need to specify the image tag you noted down earlier. Once done, click `Update Values`, and wait for your app to be refreshed.

## Step 2: Modify your build process

This step involves modifying your CI build process to ensure that instead of using the Porter CLI to simply update the Porter app. In your current build process, you'll see sections that use `porter update config` to update the image tag being used for the app; we need to add an additional step right after this update, which tells Porter to switch traffic to the new version. 

As an example, let's consider the following Github Action workflow step, that currently simply updates an app on Porter called `web-sample` with the new image tag:

```yaml
    porter-deploy-app:
        runs-on: ubuntu-latest
        steps:
            - name: Checkout code
              uses: actions/checkout@v2.3.4
            - name: Set Github tag
              id: vars
              run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"

>             - name: Update configuration for webserver-dev

Reviewer: ```suggestion
            - name: Update configuration for web-sample
```

              uses: porter-dev/porter-cli-action@v0.1.0
              with:
                  command: update config --app web-sample --tag ${{ steps.vars.outputs.sha_short }} --stream
              env:
                  PORTER_HOST: https://dashboard.getporter.dev
                  PORTER_CLUSTER: 0000
                  PORTER_PROJECT: 0000
                  PORTER_TOKEN: ${{ secrets.PORTER_TOKEN_0000 }}
```

To enable B/G deployments in this case, we'll just add an additional step at the end of this block to switch traffic, like so:

```yaml
    porter-deploy-app:
        runs-on: ubuntu-latest
        steps:
            - name: Checkout code
              uses: actions/checkout@v2.3.4
            - name: Set Github tag
              id: vars
              run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"

>             - name: Update configuration for webserver-dev

Reviewer: ```suggestion
            - name: Update configuration for web-sample
```

              uses: porter-dev/porter-cli-action@v0.1.0
              with:
                  command: update config --app web-sample --tag ${{ steps.vars.outputs.sha_short }} --stream
              env:
                  PORTER_HOST: https://dashboard.getporter.dev
                  PORTER_CLUSTER: 0000
                  PORTER_PROJECT: 0000
                  PORTER_TOKEN: ${{ secrets.PORTER_TOKEN_0000 }}

>             - name: Switch traffic to the new app once ready for frontend-dev

Reviewer: ```suggestion
            - name: Switch traffic to the new app once ready for web-sample
```

              uses: porter-dev/porter-cli-action@v0.1.0
              with:
                command: deploy blue-green-switch --app web-sample --tag ${{ steps.vars.outputs.sha_short }}
              env:
                PORTER_HOST: https://dashboard.getporter.dev
                PORTER_CLUSTER: 0000
                PORTER_PROJECT: 0000
                PORTER_TOKEN: ${{ secrets.PORTER_TOKEN_0000 }}
```

Once you've saved this and triggered a new set of builds, your CI runner will be able to specify a "green" version of your app, and switch traffic to it once the "green" version is ready. 

> :::warning Environment

Reviewer: ```suggestion
:::info Environment
```
